File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    result = []
    for i in range(0, len(queries)):
        s = queries[i] [0]
        e = queries[i] [1]
        k = queries[i] [2]
        temp = []
        for j in range(s, e+1):
            if arr[j] > k:
                temp.append(arr[j])
        if temp:
            result.append(min(temp))
        else:
            result.append(-1)
    return result

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int[] tmp = new int[queries.length];
        
        for (int j = 0; j < queries.length; j++) { // 1차원 배열 돌림
            for (int k = queries[j][0]; k <= queries[j][1]; k++) { //s <= k <= e
                
                if (answer[j] == 0 && arr[k] > queries[j][2]) {
                    answer[j] = arr[k];
                }
                else if (answer[j] != 0 && arr[k] > queries[j][2]) {
                    if (arr[k] < answer[j]) answer[j] = arr[k];
                }
            }
            if (answer[j] == 0) answer[j] = -1;
        }
        return answer;
    }
}